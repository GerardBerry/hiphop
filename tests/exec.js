"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module>
	<hh.outputsignal name="O"/>
	<hh.outputsignal name="OT" valued/>
	<hh.inputsignal name="T" valued/>
	<hh.parallel>
	  <hh.sequence>
	    <hh.exec signal="T" start=${function() {
	       console.log("Oi.");
	       setTimeout(function(self) {
		  console.log("Oi timeout.");
		  self.return(5);
	       }, 3000, this);
	    }}/>
	    <hh.emit signal="OT" arg=${hh.value("T")}/>
	  </hh.sequence>
	  <hh.emit signal="O"/>
	</hh.parallel>
      </hh.module>

var machine = new hh.ReactiveMachine(prg, "exec");

console.log(machine.ast.pretty_print());
console.log(machine.react());
console.log(machine.react());
console.log(machine.react());
console.log(".......");
setTimeout(function() {
   console.log(machine.react());
   console.log(machine.react());
}, 5000);

