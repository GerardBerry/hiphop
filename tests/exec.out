[0] +--- Module*
[0]    +--- OutputSignal O
[0]    +--- OutputSignal OT
[0]    +--- InputSignal T
[1]    +--- Parallel
[1]       +--- Sequence
[1]          +--- Exec
[1]          +--- Emit OT
[1]       +--- Emit O

Oi.
[ 'O' ]
[]
[]
.......
Oi timeout.
[ 'OT(5)' ]
[]
